### PR TITLE
Double support on all expression evaluators

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Mostly mechanical work of replacing `int64_t` -> `IntOrDouble`

Split out from https://github.com/csarofeen/pytorch/pull/1933 for easier review.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/dynamic_type.h [hunk 1]
```diff
@@ -169,21 +180,10 @@ class TORCH_CUDA_CU_API IntOrDouble {
...
 
-  template <typename T>
-  bool operator==(T val) const {
-    return operator==(IntOrDouble(val));
-  }
-
-  template <typename T>
-  bool operator!=(T val) const {
-    return operator!=(IntOrDouble(val));
-  }
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/dynamic_type.h:180]: Wasn't this just added? do we want to get rid of it?

File: torch/csrc/jit/codegen/cuda/evaluator_common.cpp [hunk 2]
```diff
@@ -290,52 +294,62 @@ void NaiveIntegerMachine<IRContext>::runInstruction(int index) {
...
+    case UnaryOpType::Set:
       dest = src;
       break;
+    case UnaryOpType::Cast:
+      if (data_type_[index] == DataType::Double) {
+        dest = src.template cast<double>();
+      } else if (data_type_[index] == DataType::Int) {
+        dest = src.template cast<int64_t>();
+      } else {
+        TORCH_INTERNAL_ASSERT(false, "wrong type");
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/evaluator_common.cpp]: NIT: Could you put something more along the lines of `dtype not supported in evaluator`?

**Review by csarofeen — APPROVED** [review]
LGTM just one real comment.

File: torch/csrc/jit/codegen/cuda/dynamic_type.h (hunk 1 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/dynamic_type.h:180]: This is covered by previous line
```c++
  DEFINE_COMPARE_OP(==)
  DEFINE_COMPARE_OP(!=)
```

File: torch/csrc/jit/codegen/cuda/evaluator_common.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/evaluator_common.cpp]: fixed

**Pull request merged**